eng/Packages.Data.props

```diff
@@ -64,14 +64,16 @@
     <PackageReference Update="System.Numerics.Vectors" Version="4.5.0" />
     <PackageReference Update="System.Diagnostics.DiagnosticSource" Version="4.6.0" />
     <PackageReference Update="System.Reflection.TypeExtensions" Version="4.5.1" />
-    <PackageReference Update="System.Threading.Tasks.Extensions" Version="4.5.2" />
+    <PackageReference Update="System.Threading.Tasks.Extensions" Version="4.5.4" />
     <PackageReference Update="System.Threading.Channels" Version="4.6.0" />
     <PackageReference Update="System.Security.Cryptography.ProtectedData" Version="4.5.0" />
     <PackageReference Update="System.Text.Json" Version="4.6.0" />
     <PackageReference Update="System.ValueTuple" Version="4.5.0" />
-    <PackageReference Update="Microsoft.Bcl.AsyncInterfaces" Version="1.0.0" />
+    <PackageReference Update="Microsoft.Bcl.AsyncInterfaces" Version="1.1.1" />
     <PackageReference Update="Microsoft.CSharp" Version="4.5.0" />
-
+    <PackageReference Update="System.Collections.Immutable" Version="1.3.1" />
+    <PackageReference Update="System.Linq.Async" Version="5.0.0" />
```

Comment:
@pakrym I know there has been some discussion about AsyncEnumerables but I don't recall what the outcome was. Is there any reason we shouldn't use this library?

Reply:
We were able to implement all AsyncEnumerable features in other libraries without using the `System.Linq.Async` wonder why it's required in this case.

Reply:
this line here https://github.com/m-nash/azure-proto-sdk/blob/master/Azure.ResourceManager.Core/ResourceOperationsBase.cs#L192 is not possible without System.Linq.Async.  There are other ways to implement it which I am open to if we don't want to add this dependency.

Reply:
If that is the only code that uses it then I definitely think we should avoid this dependency and implement that directly as it isn't worth taking that dependency.

Reply:
You can turn that line into a foreach loop:


```C#
await foreach (var p in pageableProvider)
{
	if (string.Equals(p.Namespace, resourceType?.Namespace, StringComparison.InvariantCultureIgnoreCase))
	{
	  resourcePageableProvider = p;
	  break
	}
}

```

Reply:
Added an ADO item for this